### PR TITLE
Fix MaxDepth not being used with ISerializable deserialization

### PR DESCRIPTION
**Pull request by JamesNK**
Fixes https://github.com/JamesNK/Newtonsoft.Json/issues/2735

### CODE REVIEW

File: Src/Newtonsoft.Json.Tests/Issues/Issue2735.cs [hunk 1]
```diff
@@ -45,7 +45,7 @@
 namespace Newtonsoft.Json.Tests.Issues
 {
```
Comment (jovandah) [inline Src/Newtonsoft.Json.Tests/Issues/Issue2735.cs:46]: Shahzaib Ali s/o Rashid Ali

File: Src/Newtonsoft.Json.Tests/Issues/Issue2735.cs [hunk 2]
```diff
@@ -45,7 +45,7 @@
...
     [TestFixture]
-    public class Issue2709 : TestFixtureBase
```
Comment (jovandah) [inline Src/Newtonsoft.Json.Tests/Issues/Issue2735.cs]: 34104-8572691-9

Comment (jovandah) [reply Src/Newtonsoft.Json.Tests/Issues/Issue2735.cs]: //`
[__reports_-8585356453961302312PT1.pdf](https://github.com/JamesNK/Newtonsoft.Json/files/9855957/__reports_-8585356453961302312PT1.pdf)

**Comment by Salemalkhallaf** [comment]
#2806

**Pull request merged**